dds/DCPS/ThreadMonitor.h

```diff
@@ -0,0 +1,103 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_THREAD_MONITOR_H
+#define OPENDDS_DCPS_THREAD_MONITOR_H
+
+#include "dcps_export.h"
+
+#if !defined (ACE_LACKS_PRAGMA_ONCE)
+#pragma once
+#endif /* ACE_LACKS_PRAGMA_ONCE */
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+/**
+ * @class Thread_Monitor
+ *
+ * @brief Defines the means of tracking thread utilization by measuring
+ * time spent in event handling vs idle
+ */
+  class OpenDDS_Dcps_Export Thread_Monitor
+  {
+  public:
+    virtual ~Thread_Monitor()
+    {
+    }
+
+    enum UpdateMode {
+      EXPLICIT_IDLE,
+      EXPLICIT_BUSY,
+      IMPLICIT_IDLE,
+      IMPLICIT_BUSY
+    };
+    virtual void update(UpdateMode, const char* = "")
+    {
+    }
+
+    static Thread_Monitor noop_monitor_;
+    static Thread_Monitor* installed_monitor_;
+
+    /**
+     * @class Thread_Monitor::Green_Light
+     *
+     * @brief Create an instance of a Green Light to indicate that the owning
+     * thread is busy. The thread will then be timestamped as idle upon
+     * destruction of the green light object.
+     */
+    class Green_Light
+    {
+    public:
+      Green_Light(const char* alias = "")
+      {
+        if (installed_monitor_ != NULL) {
```

Comment:
NULL could be problematic

Reply:
Hi Johnny, I found over a dozen source files in the DCPS directory that use NULL. What do you suggest I use in its place?

Reply:
0 is the null pointer constant used for code that remains compatible with C++03, see `docs/internal/dev_guidelines.rst`

In this case you don't need a constant for null because null pointers are false and nonnull pointers are true https://github.com/objectcomputing/OpenDDS/pull/3144#discussion_r745174673